myvpn-client/main.go

```diff
@@ -90,11 +122,19 @@ func main() {
 			}
 			glog.V(3).Infoln("recieve IP packet from tun", b[:n])
 
-			_, err = conn.Write(b[:n])
-			if err != nil {
-				glog.Errorln("fail to write to connection", err)
-				errc <- err
+			for {
+				_, err = conn.Write(b[:n])
```

Comment:
It may be a problem of a forever loop here, what if the error is unrecoverable? Then no matter how many times the client loops, it still can't get a success write?